python/az/aro/azext_aro/tests/latest/unit/test_validators.py

```diff
@@ -0,0 +1,419 @@
+    get_mgmt_service_client_mock.return_value = client_mock
+    if expected_exception is None:
+        validate_cluster_resource_group(cmd_mock, namespace)
+    else:
+        with pytest.raises(expected_exception):
+            validate_cluster_resource_group(cmd_mock, namespace)
+
+
+test_validate_disk_encryption_set_data = [
+    ("should not raise any exception when namespace.disk_encryption_set is None", None, Mock(disk_encryption_set=None), None, None, None, None),
+    ("should raise InvalidArgumentValueError exception when namespace.disk_encryption_set is not None and is_valid_resource_id(namespace.disk_encryption_set) returns False", None, Mock(disk_encryption_set="something different than None"), False, None, InvalidArgumentValueError, None),
+    ("should not raise any exception when compute_client.disk_encryption_sets.get() not raises CludError exception", Mock(cli_ctx=None), Mock(disk_encryption_set="something different than None"), True, Mock(), None, {"resource_group": None, "name": None})
+]
+
+
+@pytest.mark.parametrize("test_description, cmd_mock, namespace, is_valid_resource_id_return_value, compute_client_mock, expected_exception, parse_resource_id_mock_return_value", test_validate_disk_encryption_set_data, ids=[i[0] for i in test_validate_disk_encryption_set_data])
+@patch('azext_aro._validators.get_mgmt_service_client')
+@patch('azext_aro._validators.parse_resource_id')
+@patch('azext_aro._validators.is_valid_resource_id')
+def test_validate_disk_encryption_set(is_valid_resource_id_mock, parse_resource_id_mock, get_mgmt_service_client_mock, test_description, cmd_mock, namespace, is_valid_resource_id_return_value, compute_client_mock, expected_exception, parse_resource_id_mock_return_value):
+    is_valid_resource_id_mock.return_value = is_valid_resource_id_return_value
+    parse_resource_id_mock.return_value = parse_resource_id_mock_return_value
+
+    if compute_client_mock is not None:
+        compute_client_mock.get.return_value = None
+        get_mgmt_service_client_mock.return_value = compute_client_mock
+
+    if expected_exception is None:
+        validate_disk_encryption_set(cmd_mock, namespace)
+    else:
+        with pytest.raises(expected_exception):
+            validate_disk_encryption_set(cmd_mock, namespace)
+
+
+test_validate_domain_data = [
+    ("should not raise any exception when namespace.domain is None", Mock(domain=None), None),
+    ("should not raise any exception when namespace.domain has '-'", Mock(domain="my-domain.com"), None),
+    ("should not raise any exception when namespace.domain is google.com.au", Mock(domain="google.com.au"), None),
+    ("should not raise any exception when namespace.domain is some.more.than.expected", Mock(domain="some.more.than.expected"), None),
+    ("should not raise any exception when namespace.domain is azure.microsoft.com", Mock(domain="azure.microsoft.com"), None),
```

Comment:
These seem like the same (happy-path) case, can we remove the duplicates, or am I missing something?

Reply:
Done (removed test case from line 121).